### PR TITLE
Readme update : intro, installation instructions and example

### PR DESCRIPTION
**Pull request by makeitcount**
## Summary of change
* Intro
* Installation instructions
* Example
* Link to backend SDKs

## Related issues
- #114

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,45 @@
...
+# SuperTokens Javascript Frontend SDK
 
 <a href="https://supertokens.io/discord">
 <img src="https://img.shields.io/discord/603466164219281420.svg?logo=discord"
     alt="chat on Discord"></a>
-    
-## About
-This is a website / JavaScript SDK that is responsible for maintaining a SuperTokens session for website apps (SPA and non-SPA).
+
+SuperTokens adds **secure login and session management** to your apps. This is the Javascript Frontend SDK for SuperTokens. [More SDKs are available](https://github.com/supertokens) for frontend and backend e.g. Node.js, Go, Python, React.js, React Native, Vanilla JS, etc.
```
Comment (rishabhpoddar) [inline README.md:10]: Shuold mention that this only helps with session management?

File: README.md [hunk 2]
```diff
@@ -1,17 +1,45 @@
...
+
 
 Learn more at https://supertokens.io
 
+
+# How to install
+
+```
+<script src="https://cdn.jsdelivr.net/gh/supertokens/supertokens-website/bundle/bundle.js"></script>
+```
```
Comment (rishabhpoddar) [inline README.md:23]: There is also the npm way to install it. Put that too.

File: README.md [hunk 3]
```diff
@@ -1,17 +1,45 @@
...
+
+# Example
+
+```
+supertokens.init({
+    apiDomain: "<URL to your auth backend>",
+    apiBasePath: "/auth"
```
Comment (rishabhpoddar) [inline README.md:30]: You don't need to give apiBasePath

File: README.md [hunk 4]
```diff
@@ -1,17 +1,45 @@
...
+});
+// Now your app will maintain secure SuperTokens sessions for your users
```
Comment (rishabhpoddar) [inline README.md:32]: This comment is misleading. It's not the only thing that needs to be done.

You should also add a section about using with `fetch` and with `axios`. With `fetch`, nothing more needs to be done, with `axios`, you need to add interceptors.

File: README.md [hunk 5]
```diff
@@ -1,17 +1,45 @@
...
-## About
-This is a website / JavaScript SDK that is responsible for maintaining a SuperTokens session for website apps (SPA and non-SPA).
+
+SuperTokens adds **secure login and session management** to your apps. This is the Javascript Frontend SDK for SuperTokens. [More SDKs are available](https://github.com/supertokens) for frontend and backend e.g. Node.js, Go, Python, React.js, React Native, Vanilla JS, etc.
+
+![Architecture Diagram](https://supertokens.com/img/architecture/self_hosted_generic.png)
+SuperTokens architecture is optimized to add secure authentication for your users without compromising on user and developer experience
+
 
 Learn more at https://supertokens.io
```
Comment (rishabhpoddar) [inline README.md:16]: Change to supertokens.com

File: README.md [hunk 6]
```diff
@@ -1,17 +1,45 @@
...
+
+# How to install
+
+```
+<script src="https://cdn.jsdelivr.net/gh/supertokens/supertokens-website/bundle/bundle.js"></script>
+```
+
+# Example
+
+```
```
Comment (rishabhpoddar) [inline README.md:27]:
```suggestion
```js
```

File: README.md [hunk 7]
```diff
@@ -1,17 +1,45 @@
...
 Learn more at https://supertokens.io
 
+
+# How to install
+
+```
+<script src="https://cdn.jsdelivr.net/gh/supertokens/supertokens-website/bundle/bundle.js"></script>
+```
+
+# Example
```
Comment (rishabhpoddar) [inline README.md:25]:
```suggestion
# Initialising
```

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (gitcommitshow) [reply README.md:10]: What do users need to do to implement login for their vanilla js website? (when they already have supertokens backend)

Comment (rishabhpoddar) [reply README.md:10]: We are working on that SDK (https://github.com/supertokens/supertokens-web-js) - it's not released yet.

But right now, they would use supertokens-website for sessions and call the APIs using `axios` or `fetch` for login.

File: README.md (hunk 2 above)
Comment (gitcommitshow) [reply README.md:23]: are there significant no. of people who install frontend library via npm?

I want to keep the instructions minimal. And have another link for detailed documentation.

File: README.md (hunk 4 above)
Comment (gitcommitshow) [reply README.md:32]: we should add an example using fetch

File: README.md (hunk 7 above)
Comment (gitcommitshow) [reply README.md:25]: I'm not sure, why?

File: README.md (hunk 2 above)
Comment (rishabhpoddar) [reply README.md:23]: Most would use npm

File: README.md (hunk 7 above)
Comment (rishabhpoddar) [reply README.md:25]: Cause an example would be making an request. The contents of this section is about initialising the SDK

Comment (makeitcount) [reply README.md]: New changes should convey this as well as keep the heading simple

File: README.md (hunk 6 above)
Comment (makeitcount) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (makeitcount) [reply README.md:33]: removed apiBasePath. Will add info about `fetch` code block once I learn more about login endpoints. We should skip mentioning `axios` here IMO.

File: README.md (hunk 2 above)
Comment (makeitcount) [reply README.md]: added npm method

File: README.md [hunk 8]
```diff
@@ -22,16 +22,17 @@ Learn more at https://supertokens.io
 <script src="https://cdn.jsdelivr.net/gh/supertokens/supertokens-website/bundle/bundle.js"></script>
 ```
 
-# Example
+# How to use
 
 ```
```
Comment (rishabhpoddar) [inline README.md:31]:
```suggestion
```js
```

File: README.md [hunk 9]
```diff
@@ -22,16 +22,17 @@ Learn more at https://supertokens.io
...
+// Initialization
 supertokens.init({
-    apiDomain: "<URL to your auth backend>",
-    apiBasePath: "/auth"
+    apiDomain: "<URL to your auth backend>"
 });
 // Now your app will maintain secure SuperTokens sessions for your users
-```
+```js
```
Comment (rishabhpoddar) [inline README.md]:
```suggestion
```
```

File: README.md (hunk 4 above)
Comment (rishabhpoddar) [reply README.md:41]: We shouldn't skip mentioning axios. A LOT of people use axios.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -1,16 +1,52 @@
...
+
+// Now your app will maintain secure SuperTokens sessions for your users
+```
+2. Make sure your backend has the needed auth functionalities
+> You can use one of the SuperTokens backend SDKs for this. 
+> Backend SDKs are available for
+> * [Node.js backend](https://github.com/supertokens/supertokens-node)
+> * [Golang backend](https://github.com/supertokens/supertokens-golang)
+> * [Python backend](https://github.com/supertokens/supertokens-python)
+
```
Comment (rishabhpoddar) [inline README.md:49]: Please also update the supertokens.io link to supertokens.com everywhere else in this PR (including for the email ID at the bottom)

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Comment by makeitcount** [comment]
@rishabhpoddar [this doc](https://supertokens.com/docs/website/installation) seems to be more complete and useful, I am thinking of linking it in README. But it is for `8.2.x` and not sure if there are major breaking changes or not. Should we link it?

File: README.md (hunk 4 above)
Comment (makeitcount) [reply README.md:41]: done. mentioned the complete axios example

**Comment by makeitcount** [comment]
@rishabhpoddar please review and close if possible. So we can see the impact this update makes. We will come back to refactoring again as we have some more pending work items for README upgrade.

**Comment by rishabhpoddar** [comment]
hey @makeitcount the contents in the readme, specifically to do with get jwt payload is wrong.

Please make sure you have taken the content from the right version of the SDK docs. I won't be merging this until you have done this properly.

**Pull request merged**